modules/ossm-integrating-with-user-workload-monitoring.adoc

```diff
@@ -0,0 +1,174 @@
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    enableUserWorkload: true
+----
+
+. Configure the Service Mesh control plane for User-Workload Monitoring:
++
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic
+  namespace: istio-system
+spec:
+  addons:
+    prometheus:
+      enabled: false # <1>
+    grafana:
+      enabled: false # <2>
+  meshConfig:
+    extensionProviders:
+    - name: prometheus # <3>
+----
+<1> Disable the default Prometheus instance provided by OSSM.
+<2> Disable Grafana. It is not supported with an external Prometheus instance.
+<3> Configure the Prometheus extension provider.
++
+
+. Apply a custom network policy to allow ingress traffic from the `user-workload monitoring` namespace:
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: user-workload-access
+  namespace: bookinfo
+spec:
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+          network.openshift.io/policy-group: monitoring
+  podSelector: {}
+  policyTypes:
+  - Ingress
+----
++
+
+. Apply a telemetry object to enable traffic metrics in Istio proxies:
++
+[source,yaml]
+----
+apiVersion: telemetry.istio.io/v1alpha1
+kind: Telemetry
+metadata:
+  name: enable-prometheus-metrics
+  namespace: istio-system # <1>
+spec:
+  selector: # <2>
+    matchLabels:
+      app: bookinfo 
+  metrics:
+  - providers:
+    - name: prometheus
+----
+<1> A telemetry resource created in the control plane namespace applies to all workloads in a mesh. To apply telemetry to only one namespace, create the resource in the target namespace.
+<2> (Optional) Setting `spec.selector.matchLabels` applies the telemetry resource to specific workloads in the target namespace.
++
+
+. Apply a `ServiceMonitor` to monitor the Istio control plane:
++
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: istiod-monitor
+  namespace: istio-system # <1>
+spec:
+  targetLabels:
+  - app
+  selector:
+    matchLabels:
+      istio: pilot
+  endpoints:
+  - port: http-monitoring
+    interval: 30s
+    relabelings:
+    - action: replace
+      replacement: "<SMCP_NAME>-<SMCP_NAMESPACE>" # <2>
+      targetLabel: mesh_id
+----
+<1> Since OpenShift Monitoring ignores the field `spec.namespaceSelector` in `ServiceMonitor` and `PodMonitor` objects, you need to apply the `PodMonitor` in all mesh namespaces, including the control plane namespace.
+<2> A shared Prometheus instance can mean there are metrics from multiple meshes. However, if you are deploying Kiali, the metrics must be associated with a single mesh. To associate metrics with a single mesh, add a unique `mesh_id` to each one. Doing so narrows the `query_scope` in Kiali to only the relevant mesh metrics.
```

Comment:
```suggestion
<2> A shared Prometheus instance can include metrics from multiple meshes. However, if you are deploying Kiali, the metrics must be associated with a single mesh. To associate metrics with a single mesh, add a unique `mesh_id` to each one. Doing so narrows the query scope in Kiali to only the relevant mesh metrics.
```
Unless there's a need to include `query_scope` here I feel like it's easier to read when written as "query scope", because it makes the meaning more clear?
Just some suggestions for readability